kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/EngineUIProxyServlet.scala

```diff
@@ -30,14 +30,15 @@ private[api] class EngineUIProxyServlet extends ProxyServlet with Logging {
   override def rewriteTarget(request: HttpServletRequest): String = {
     val requestURL = request.getRequestURL
     val requestURI = request.getRequestURI
+    val queryString = getQueryString(request)
     var targetURL = "/no-ui-error"
     extractTargetAddress(requestURI).foreach { case (host, port) =>
       val targetURI = requestURI.stripPrefix(s"/engine-ui/$host:$port") match {
         // for some reason, the proxy can not handle redirect well, as a workaround,
         // we simulate the Spark UI redirection behavior and forcibly rewrite the
         // empty URI to the Spark Jobs page.
         case "" | "/" => "/jobs/"
-        case path => path
+        case path => path + queryString
```

Comment:
jobs also need queryString

Reply:
OK. will add.

Reply:
done

Reply:
We can move those `append logic` out. Like:
``` scala
val targetURI = requestURI.stripPrefix(s"/engine-ui/$host:$port") match {
    case "" | "/" => "/jobs/" 
    case path => path
} + queryString
```

Reply:
Done. Thanks.